### PR TITLE
Netty BOM inherit from parent and its own dependency management

### LINKED ISSUE #6674: Netty BOM inherit from parent and its own dependency management
A BOM was recently added in #5994 but is not valid.

A BOM should only expose dependency management for the artifacts that the project produces (so in this case, netty artifacts). It should not have an opinion about third party dependencies (otherwise including that BOM in a project potentially overrides decisions that were made about them).

One way to fix that would be to stop inheriting from `netty-parent`. Also, I'd set the version of each dependency in the bom rather than using `${project.version}`. If you're including the bom for netty X.Y.Z, it should be X.Y.Z. and nothing else (Maven does not allow to override that property, which is good so it's not strictly necessary but the intent is more clear if you hardcode the current version).

Here is an example of BOM for reference: http://repo.spring.io/release/org/springframework/spring-framework-bom/4.3.8.RELEASE/spring-framework-bom-4.3.8.RELEASE.pom

We'd like to [use this bom in Spring Boot](https://github.com/spring-projects/spring-boot/issues/8553) but we can't for that reason.

### LINKED ISSUE #6677: Netty bom does not provide dependency management for `netty-all`
The netty BOM should provide dependency management for all netty artifacts. I don't see `netty-all` in the list. Can you please add it?

### PR DESCRIPTION
**Pull request by johnou**
Motivation:

A BOM should only expose dependency management for the artifacts that the project produces (so in this case, netty artifacts). It should not have an opinion about third party dependencies (otherwise including that BOM in a project potentially overrides decisions that were made about them).

Modifications:

Stop inheriting from netty-parent and explicitly set the versions for the artifacts.

Result:

Valid BOM.

Fixes #6674, fixes #6677.

### CODE REVIEW

**Comment by johnou** [comment]
@snicoll PTAL.

File: bom/pom.xml [hunk 1]
```diff
@@ -15,15 +15,17 @@
...
+    <groupId>org.sonatype.oss</groupId>
+    <artifactId>oss-parent</artifactId>
+    <version>7</version>
+    <relativePath />
   </parent>
 
+  <groupId>io.netty</groupId>
   <artifactId>netty-bom</artifactId>
+  <version>4.1.10.Final-SNAPSHOT</version>
   <packaging>pom</packaging>
```
Comment (snicoll) [inline bom/pom.xml]: According to [this page](https://maven.apache.org/guides/mini/guide-central-repository-upload.html) you need to add the `<url>`, `<license>` and `<scm>` tags to be able to deploy it to Maven central. Check the sample BOM in the related issue for reference.

Comment (johnou) [inline bom/pom.xml]: roger.

**Comment by johnou** [comment]
@snicoll amended commit, thanks for checking it out 👍

**Comment by normanmaurer** [comment]
@snicoll does this look correct to you ?

**Comment by snicoll** [comment]
@normanmaurer looks good to me. Thanks!

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (0123190214c59d764a6d3c79a9570cd20fe6715a) and 4.0 (8e928375c1)

@johnou thanks!

**Pull request closed without merging**